schema_based_on_diagram.sql

```diff
@@ -0,0 +1,40 @@
+CREATE TABLE patients(
+	id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	name VARCHAR(255),
+	date_of_birth DATE,
+);
+
+CREATE TABLE medical_histories(
+	id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	admitted_at TIMESTAMP,
+	patient_id INT REFERENCES patients(id),
+	status VARCHAR(255)
+);
+
+CREATE TABLE treatments(
+	id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	type VARCHAR(250),
+	name VARCHAR(250)
+);
+
+CREATE TABLE medical_histories_and_treatments(
+	treatment_id INT REFERENCES treatments(id),
+	medical_histories_id INT REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items(
+	id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	unit_price DECIMAL(8,2),
+	quantity INT,
+	total_price DECIMAL(8,2),
+	treatment_id INT REFERENCES treatments(id)
+);
+
+CREATE TABLE invoices(
+	id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	total_amount DECIMAL(8,2),
+	generated_at TIMESTAMP,
+	payed_at TIMESTAMP,
+	medical_history_id INT REFERENCES medical_histories(id)
+);
+
```

Comment:
- [ ] Kindly create indexes for all foreign keys as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.).